File: examples/governance/compliance_domain/lib/compliance_domain/governance_policy/commands/suspend_policy.rb
module ComplianceDomain
  class GovernancePolicy
    module Commands
      class SuspendPolicy
        include Hecks::Command
        emits "SuspendedPolicy"

        attr_reader :policy_id

        def initialize(policy_id: nil)
          @policy_id = policy_id
        end

        def call
          existing = repository.find(policy_id)
          if existing
            unless existing.status == "active"
              raise Hecks::Error, "Cannot SuspendPolicy: status must be 'active', got '#{existing.status}'"
            end
            GovernancePolicy.new(
              id: existing.id,
              name: existing.name,
              description: existing.description,
              category: existing.category,
              framework_id: existing.framework_id,
              effective_date: existing.effective_date,
              review_date: existing.review_date,
              requirements: existing.requirements,
              status: "suspended"
            )
          else
            raise Hecks::Error, "GovernancePolicy not found: #{policy_id}"
          end
        end
      end
    end
  end
end

File: examples/governance_static_go/domain/suspend_policy.go
package domain

import (
	"time"
	"fmt"
)

type SuspendPolicy struct {
	PolicyId string `json:"policy_id"`
}

func (c SuspendPolicy) CommandName() string { return "SuspendPolicy" }

func (c SuspendPolicy) Execute(repo GovernancePolicyRepository) (*GovernancePolicy, *SuspendedPolicy, error) {
	existing, err := repo.Find(c.PolicyId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("GovernancePolicy not found: %s", c.PolicyId)
	}
	if existing.Status != "active" {
		return nil, nil, fmt.Errorf("cannot SuspendPolicy: GovernancePolicy is in %s state", existing.Status)
	}
	existing.Status = "suspended"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := SuspendedPolicy{
		AggregateID: existing.ID,
		PolicyId: c.PolicyId,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}
